diff --git a/xLights/BitmapCache.cpp b/xLights/BitmapCache.cpp
index a58a68a30..567128793 100644
--- a/xLights/BitmapCache.cpp
+++ b/xLights/BitmapCache.cpp
@@ -601,7 +601,7 @@ wxBitmapBundle BitmapCache::GetModelGroupIcon() {
     return CreateBitmapBundleFromXPMs(16, "ModelGroup", {model_16, model_64, model_64, model_64, model_64});
 }
 wxBitmapBundle BitmapCache::GetFPPIcon() {
-    return wxBitmapBundle::FromSVG((const char*)fpp_app_icon_svg, wxSize(16, 16));
+    return wxBitmapBundle::FromSVG(fpp_app_icon_svg, sizeof(fpp_app_icon_svg), wxSize(16, 16));
 }
 
 const wxImage &BitmapCache::GetCornerIcon(int position, int size) {

