diff --git a/xLights/xLightsApp.cpp b/xLights/xLightsApp.cpp
index 534ae4d29..916fa77d5 100644
--- a/xLights/xLightsApp.cpp
+++ b/xLights/xLightsApp.cpp
@@ -436,7 +436,7 @@ bool xLightsApp::OnInit()
     static log4cpp::Category &logger_base = log4cpp::Category::getInstance(std::string("log_base"));
     logger_base.info("******* OnInit: XLights started.");
 #ifdef __WXMSW__
-    MSWEnableDarkMode();
+    //MSWEnableDarkMode();
 #endif
 #if wxUSE_GLCANVAS_EGL
     // this is only needed if using the EGL canvas as it's necessary to initialize the
