diff --git a/xLights/ModelFaceDialog.cpp b/xLights/ModelFaceDialog.cpp
index 1fc8793b2..9937e1b71 100644
--- a/xLights/ModelFaceDialog.cpp
+++ b/xLights/ModelFaceDialog.cpp
@@ -175,7 +175,6 @@ ModelFaceDialog::ModelFaceDialog(wxWindow* parent, OutputManager* outputManager,
 	SingleNodeGrid->SetMinSize(wxDLG_UNIT(CoroPanel,wxSize(-1,200)));
 	SingleNodeGrid->EnableEditing(true);
 	SingleNodeGrid->EnableGridLines(true);
-	SingleNodeGrid->SetColLabelSize(20);
 	SingleNodeGrid->SetRowLabelSize(150);
 	SingleNodeGrid->SetDefaultColSize(200, true);
 	SingleNodeGrid->SetColLabelValue(0, _("Nodes"));
@@ -222,7 +221,6 @@ ModelFaceDialog::ModelFaceDialog(wxWindow* parent, OutputManager* outputManager,
 	NodeRangeGrid->SetMinSize(wxDLG_UNIT(NodeRangePanel,wxSize(-1,200)));
 	NodeRangeGrid->EnableEditing(true);
 	NodeRangeGrid->EnableGridLines(true);
-	NodeRangeGrid->SetColLabelSize(20);
 	NodeRangeGrid->SetRowLabelSize(150);
 	NodeRangeGrid->SetDefaultColSize(200, true);
 	NodeRangeGrid->SetColLabelValue(0, _("Nodes"));
@@ -289,7 +287,6 @@ ModelFaceDialog::ModelFaceDialog(wxWindow* parent, OutputManager* outputManager,
 	MatrixModelsGrid->SetMinSize(wxDLG_UNIT(Matrix,wxSize(-1,200)));
 	MatrixModelsGrid->EnableEditing(true);
 	MatrixModelsGrid->EnableGridLines(true);
-	MatrixModelsGrid->SetColLabelSize(20);
 	MatrixModelsGrid->SetRowLabelSize(150);
 	MatrixModelsGrid->SetDefaultColSize(200, true);
 	MatrixModelsGrid->SetColLabelValue(0, _("Eyes Open"));
@@ -385,6 +382,10 @@ ModelFaceDialog::ModelFaceDialog(wxWindow* parent, OutputManager* outputManager,
 
     FaceTypeChoice->ChangeSelection(NODE_RANGE_FACE);
 
+    MatrixModelsGrid->SetColLabelSize(int(1.5 * float(MatrixModelsGrid->GetLabelFont().GetPixelSize().y)));
+    NodeRangeGrid->SetColLabelSize(int(1.5 * float(NodeRangeGrid->GetLabelFont().GetPixelSize().y)));
+    SingleNodeGrid->SetColLabelSize(int(1.5 * float(SingleNodeGrid->GetLabelFont().GetPixelSize().y)));
+
     FlexGridSizer1->Fit(this);
     FlexGridSizer1->SetSizeHints(this);
     Center();
diff --git a/xLights/ModelStateDialog.cpp b/xLights/ModelStateDialog.cpp
index aff536d26..7516679ae 100644
--- a/xLights/ModelStateDialog.cpp
+++ b/xLights/ModelStateDialog.cpp
@@ -161,7 +161,6 @@ ModelStateDialog::ModelStateDialog(wxWindow* parent, OutputManager* outputManage
 	SingleNodeGrid->SetMinSize(wxDLG_UNIT(CoroPanel,wxSize(-1,200)));
 	SingleNodeGrid->EnableEditing(true);
 	SingleNodeGrid->EnableGridLines(true);
-	SingleNodeGrid->SetColLabelSize(20);
 	SingleNodeGrid->SetDefaultColSize(150, true);
 	SingleNodeGrid->SetColLabelValue(0, _("State"));
 	SingleNodeGrid->SetColLabelValue(1, _("Nodes"));
@@ -190,7 +189,6 @@ ModelStateDialog::ModelStateDialog(wxWindow* parent, OutputManager* outputManage
 	NodeRangeGrid->SetMinSize(wxDLG_UNIT(NodeRangePanel,wxSize(-1,200)));
 	NodeRangeGrid->EnableEditing(true);
 	NodeRangeGrid->EnableGridLines(true);
-	NodeRangeGrid->SetColLabelSize(20);
 	NodeRangeGrid->SetDefaultColSize(150, true);
 	NodeRangeGrid->SetColLabelValue(0, _("State"));
 	NodeRangeGrid->SetColLabelValue(1, _("Nodes"));
@@ -258,6 +256,9 @@ ModelStateDialog::ModelStateDialog(wxWindow* parent, OutputManager* outputManage
 
     StateTypeChoice->ChangeSelection(NODE_RANGE_STATE);
 
+    NodeRangeGrid->SetColLabelSize(int(1.5 * float(NodeRangeGrid->GetLabelFont().GetPixelSize().y)));
+    SingleNodeGrid->SetColLabelSize(int(1.5 * float(SingleNodeGrid->GetLabelFont().GetPixelSize().y)));
+
     FlexGridSizer1->Fit(this);
     FlexGridSizer1->SetSizeHints(this);
     Center();
diff --git a/xLights/SubModelsDialog.cpp b/xLights/SubModelsDialog.cpp
index f04f03f3b..24844de55 100644
--- a/xLights/SubModelsDialog.cpp
+++ b/xLights/SubModelsDialog.cpp
@@ -242,7 +242,6 @@ SubModelsDialog::SubModelsDialog(wxWindow* parent, OutputManager* om) :
 	NodesGrid->CreateGrid(5,1);
 	NodesGrid->EnableEditing(true);
 	NodesGrid->EnableGridLines(true);
-	NodesGrid->SetColLabelSize(18);
 	NodesGrid->SetDefaultColSize(160, true);
 	NodesGrid->SetColLabelValue(0, _("Node Ranges"));
 	NodesGrid->SetRowLabelValue(0, _("Strand   1"));
@@ -258,19 +257,19 @@ SubModelsDialog::SubModelsDialog(wxWindow* parent, OutputManager* om) :
 	FlexGridSizer4->Add(FlexGridSizer8, 1, wxALL|wxEXPAND, 5);
 	FlexGridSizer5 = new wxFlexGridSizer(0, 1, 0, 0);
 	AddRowButton = new wxButton(Panel1, ID_BUTTON1, _("Add Row"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON1"));
-	FlexGridSizer5->Add(AddRowButton, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(AddRowButton, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	DeleteRowButton = new wxButton(Panel1, ID_BUTTON2, _("Delete Row"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON2"));
-	FlexGridSizer5->Add(DeleteRowButton, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(DeleteRowButton, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	Button_MoveUp = new wxButton(Panel1, ID_BUTTON_MOVE_UP, _("^"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON_MOVE_UP"));
-	FlexGridSizer5->Add(Button_MoveUp, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(Button_MoveUp, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	Button_MoveDown = new wxButton(Panel1, ID_BUTTON_MOVE_DOWN, _("v"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON_MOVE_DOWN"));
-	FlexGridSizer5->Add(Button_MoveDown, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(Button_MoveDown, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	Button_ReverseRow = new wxButton(Panel1, ID_BUTTON7, _("Reverse Row"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON7"));
-	FlexGridSizer5->Add(Button_ReverseRow, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(Button_ReverseRow, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	Button_SortRow = new wxButton(Panel1, ID_BUTTON_SORT_ROW, _("Sort Row"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON_SORT_ROW"));
-	FlexGridSizer5->Add(Button_SortRow, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(Button_SortRow, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	Button_Draw_Model = new wxButton(Panel1, ID_BUTTON_DRAW_MODEL, _("Draw Model"), wxDefaultPosition, wxDefaultSize, 0, wxDefaultValidator, _T("ID_BUTTON_DRAW_MODEL"));
-	FlexGridSizer5->Add(Button_Draw_Model, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL, 5);
+	FlexGridSizer5->Add(Button_Draw_Model, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 5);
 	FlexGridSizer4->Add(FlexGridSizer5, 1, wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE, 0);
 	Panel1->SetSizer(FlexGridSizer4);
 	FlexGridSizer4->Fit(Panel1);
@@ -351,6 +350,8 @@ SubModelsDialog::SubModelsDialog(wxWindow* parent, OutputManager* om) :
 	Connect(wxID_ANY,wxEVT_INIT_DIALOG,(wxObjectEventFunction)&SubModelsDialog::OnInit);
 	//*)
 
+    NodesGrid->SetColLabelSize(int(1.5 * float(NodesGrid->GetLabelFont().GetPixelSize().y)));
+
     Connect(ID_NOTEBOOK1, wxEVT_NOTEBOOK_PAGE_CHANGED, (wxObjectEventFunction)& SubModelsDialog::OnTypeNotebookPageChanged);
     Connect(wxID_ANY, EVT_SMDROP, (wxObjectEventFunction)&SubModelsDialog::OnDrop);
     Connect(ID_GRID1, wxEVT_GRID_CELL_CHANGED,(wxObjectEventFunction)&SubModelsDialog::OnNodesGridCellChange);
diff --git a/xLights/wxsmith/ModelFaceDialog.wxs b/xLights/wxsmith/ModelFaceDialog.wxs
index 20798d2df..c55f337ec 100644
--- a/xLights/wxsmith/ModelFaceDialog.wxs
+++ b/xLights/wxsmith/ModelFaceDialog.wxs
@@ -107,7 +107,6 @@
 															<object class="wxGrid" name="ID_GRID_COROFACES" variable="SingleNodeGrid" member="yes">
 																<cols>2</cols>
 																<rows>13</rows>
-																<labelrowheight>20</labelrowheight>
 																<labelcolwidth>150</labelcolwidth>
 																<defaultcolsize>200</defaultcolsize>
 																<collabels>
@@ -196,7 +195,6 @@
 															<object class="wxGrid" name="ID_GRID3" variable="NodeRangeGrid" member="yes">
 																<cols>2</cols>
 																<rows>28</rows>
-																<labelrowheight>20</labelrowheight>
 																<labelcolwidth>150</labelcolwidth>
 																<defaultcolsize>200</defaultcolsize>
 																<collabels>
@@ -317,7 +315,6 @@
 															<object class="wxGrid" name="ID_GRID1" variable="MatrixModelsGrid" member="yes">
 																<cols>2</cols>
 																<rows>10</rows>
-																<labelrowheight>20</labelrowheight>
 																<labelcolwidth>150</labelcolwidth>
 																<defaultcolsize>200</defaultcolsize>
 																<collabels>
diff --git a/xLights/wxsmith/ModelStateDialog.wxs b/xLights/wxsmith/ModelStateDialog.wxs
index dc3e17d75..16cb82f49 100644
--- a/xLights/wxsmith/ModelStateDialog.wxs
+++ b/xLights/wxsmith/ModelStateDialog.wxs
@@ -117,7 +117,6 @@
 													<object class="wxGrid" name="ID_GRID_COROSTATES" variable="SingleNodeGrid" member="yes">
 														<cols>3</cols>
 														<rows>200</rows>
-														<labelrowheight>20</labelrowheight>
 														<labelcolwidth>0</labelcolwidth>
 														<defaultcolsize>150</defaultcolsize>
 														<collabels>
@@ -186,7 +185,6 @@
 													<object class="wxGrid" name="ID_GRID3" variable="NodeRangeGrid" member="yes">
 														<cols>3</cols>
 														<rows>200</rows>
-														<labelrowheight>20</labelrowheight>
 														<labelcolwidth>0</labelcolwidth>
 														<defaultcolsize>150</defaultcolsize>
 														<collabels>
diff --git a/xLights/wxsmith/SubModelsDialog.wxs b/xLights/wxsmith/SubModelsDialog.wxs
index 37dd93f4b..6d9d9114b 100644
--- a/xLights/wxsmith/SubModelsDialog.wxs
+++ b/xLights/wxsmith/SubModelsDialog.wxs
@@ -223,7 +223,6 @@
 																	<object class="wxGrid" name="ID_GRID1" variable="NodesGrid" member="yes">
 																		<cols>1</cols>
 																		<rows>5</rows>
-																		<labelrowheight>18</labelrowheight>
 																		<defaultcolsize>160</defaultcolsize>
 																		<collabels>
 																			<item>Node Ranges</item>
@@ -281,7 +280,7 @@
 																		<label>Add Row</label>
 																		<handler function="OnAddRowButtonClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -290,7 +289,7 @@
 																		<label>Delete Row</label>
 																		<handler function="OnDeleteRowButtonClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -299,7 +298,7 @@
 																		<label>^</label>
 																		<handler function="OnButton_MoveUpClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -308,7 +307,7 @@
 																		<label>v</label>
 																		<handler function="OnButton_MoveDownClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -317,7 +316,7 @@
 																		<label>Reverse Row</label>
 																		<handler function="OnButton_ReverseRowClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -326,7 +325,7 @@
 																		<label>Sort Row</label>
 																		<handler function="OnButton_SortRowClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
@@ -335,7 +334,7 @@
 																		<label>Draw Model</label>
 																		<handler function="OnButton_Draw_ModelClick" entry="EVT_BUTTON" />
 																	</object>
-																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+																	<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL|wxFIXED_MINSIZE</flag>
 																	<border>5</border>
 																	<option>1</option>
 																</object>
