diff --git a/xLights/controllers/FPP.cpp b/xLights/controllers/FPP.cpp
index b6cebd76c..bfb65f5f9 100644
--- a/xLights/controllers/FPP.cpp
+++ b/xLights/controllers/FPP.cpp
@@ -602,14 +602,6 @@ bool FPP::IsDrive() {
 }
 
 bool FPP::IsVersionAtLeast(uint32_t maj, uint32_t min, uint32_t patch) const{
-    static bool hasWarned = false;
-    if (majorVersion < 6 && !hasWarned) {
-        hasWarned = true;
-        wxMessageBox("Uploading configuration and/or sequences to FPP instances less than FPP 6.x will soon be removed.  Please update FPP to the latest version.",
-                     "FPP Version Deprecated",
-                     wxICON_INFORMATION | wxCENTER | wxOK);
-    }
-    
     if (majorVersion < maj) {
         return false;
     }
