diff --git a/xLights/ScriptsDialog.cpp b/xLights/ScriptsDialog.cpp
index b0cf12bc0..3fd27ffa6 100644
--- a/xLights/ScriptsDialog.cpp
+++ b/xLights/ScriptsDialog.cpp
@@ -290,8 +290,8 @@ void ScriptsDialog::Run_Python_Script(wxString const& filepath) const
 
 void ScriptsDialog::OnButton_DownloadClick(wxCommandEvent& event)
 {
-    //https://api.github.com/repos/xLightsSequencer/xLights/contents/scripts
-    std::string json_data = Curl::HTTPSGet(R"(https://api.github.com/repos/xLightsSequencer/xLights/contents/scripts)");
+    //https://api.github.com/repos/XLIGHTS_GH_BRANCH/contents/scripts
+    std::string json_data = Curl::HTTPSGet(R"(https://api.github.com/repos/" XLIGHTS_GH_BRANCH "/contents/scripts)");
     std::vector<std::pair<wxString, wxString>> scripts = std::vector<std::pair<wxString, wxString>>();
     wxJSONValue val;
     wxJSONReader reader;
diff --git a/xLights/ShaderDownloadDialog.cpp b/xLights/ShaderDownloadDialog.cpp
index d3e49ad23..0a1226dda 100644
--- a/xLights/ShaderDownloadDialog.cpp
+++ b/xLights/ShaderDownloadDialog.cpp
@@ -31,6 +31,8 @@
 #include "ExternalHooks.h"
 #include <log4cpp/Category.hh>
 
+#include "xLightsVersion.h"
+
 #include "utils/Curl.h"
 #undef min
 #undef max
@@ -359,7 +361,7 @@ wxXmlDocument* ShaderDownloadDialog::GetXMLFromURL(wxURI url, std::string& filen
 
 bool ShaderDownloadDialog::LoadTree(wxProgressDialog* prog, int low, int high)
 {
-    const std::string shaderlink = "https://raw.githubusercontent.com/xLightsSequencer/xLights/master/download/shaders.xml";
+    const std::string shaderlink = "https://raw.githubusercontent.com/" XLIGHTS_GH_BRANCH "/master/download/shaders.xml";
 
     bool rc = true;
     std::string filename;
diff --git a/xLights/TipOfTheDayDialog.cpp b/xLights/TipOfTheDayDialog.cpp
index 2175f9320..2bf7f43e7 100644
--- a/xLights/TipOfTheDayDialog.cpp
+++ b/xLights/TipOfTheDayDialog.cpp
@@ -22,14 +22,15 @@
 #include "xLightsMain.h"
 #include "utils/Curl.h"
 #include "CachedFileDownloader.h"
+#include "xLightsVersion.h"
 
 #include <log4cpp/Category.hh>
 
 #ifdef USE_WEBVIEW_FOR_TOD
 #include <wx/webview.h>
-static const std::string TOD_BASE_URL = "https://raw.githack.com/xLightsSequencer/xLights/master/TipOfDay/";
+static const std::string TOD_BASE_URL = "https://raw.githack.com/" XLIGHTS_GH_BRANCH "/master/TipOfDay/";
 #else
-static const std::string TOD_BASE_URL = "https://raw.githubusercontent.com/xLightsSequencer/xLights/master/TipOfDay/";
+static const std::string TOD_BASE_URL = "https://raw.githubusercontent.com/" XLIGHTS_GH_BRANCH  "/master/TipOfDay/";
 #endif
 
 #define USE_GITHUB_HOSTED_TOD
diff --git a/xLights/VendorModelDialog.cpp b/xLights/VendorModelDialog.cpp
index de4d06e43..539449cd2 100644
--- a/xLights/VendorModelDialog.cpp
+++ b/xLights/VendorModelDialog.cpp
@@ -27,6 +27,7 @@
 #include "CachedFileDownloader.h"
 #include "UtilFunctions.h"
 #include "ExternalHooks.h"
+#include "xLightsVersion.h"
 
 class MModel;
 
@@ -838,10 +839,9 @@ wxXmlDocument* VendorModelDialog::GetXMLFromURL(wxURI url, std::string& filename
 bool VendorModelDialog::LoadTree(wxProgressDialog* prog, int low, int high)
 {
     static log4cpp::Category& logger_base = log4cpp::Category::getInstance(std::string("log_base"));
-
-    const std::string vendorlink = "https://raw.githubusercontent.com/xLightsSequencer/xLights/master/download/xlights_vendors.xml";
+    const std::string vendorlink = "https://github.com/" XLIGHTS_GH_BRANCH "/raw/master/download/xlights_vendors.xml";
     const std::string vendorlinkbackup = "https://nutcracker123.com/xlights/vendors/xlights_vendors.xml";
-    //const std::string vendorlink = "http://localhost/xlights_vendors.xml";
+    // const std::string vendorlink = "http://localhost/xlights_vendors.xml";
 
     std::string filename;
     if (prog != nullptr)
diff --git a/xLights/VendorMusicDialog.cpp b/xLights/VendorMusicDialog.cpp
index 43939d374..8c3eaddbc 100644
--- a/xLights/VendorMusicDialog.cpp
+++ b/xLights/VendorMusicDialog.cpp
@@ -21,6 +21,7 @@
 #include "CachedFileDownloader.h"
 #include "UtilFunctions.h"
 #include "xLightsMain.h"
+#include "xLightsVersion.h"
 
 CachedFileDownloader& VendorMusicDialog::GetCache() {
     return CachedFileDownloader::GetDefaultCache();
@@ -211,8 +212,8 @@ wxXmlDocument* VendorMusicDialog::GetXMLFromURL(wxURI url, std::string& filename
 
 bool VendorMusicDialog::LoadTree(std::string hash)
 {
-    const std::string vendorlink = "https://nutcracker123.com/xlights/vendors/xlights_vendors.xml";
-    const std::string vendorlinkbackup = "https://github.com/xLightsSequencer/xLights/raw/master/download/xlights_vendors.xml";
+    const std::string vendorlink = "https://github.com/" XLIGHTS_GH_BRANCH "/raw/master/download/xlights_vendors.xml";
+    const std::string vendorlinkbackup = "https://nutcracker123.com/xlights/vendors/xlights_vendors.xml";
     //const std::string vendorlink = "http://127.0.0.1:3000/xlights_vendors.xml";
 
     std::string filename;
diff --git a/xLights/models/Model.cpp b/xLights/models/Model.cpp
index 24c7922c2..675f255c5 100644
--- a/xLights/models/Model.cpp
+++ b/xLights/models/Model.cpp
@@ -5998,7 +5998,7 @@ Model* Model::GetXlightsModel(Model* model, std::string& last_model, xLightsFram
                     // the user has decided they dont want it then treat them like an adult
                     if (!xlights->GetIgnoreVendorModelRecommendations()) {
 #endif
-                        wxURI mappingJson("https://raw.githubusercontent.com/xLightsSequencer/xLights/master/download/model_vendor_mapping.json");
+                        wxURI mappingJson("https://raw.githubusercontent.com/" XLIGHTS_GH_BRANCH "/master/download/model_vendor_mapping.json");
                         std::string json = CachedFileDownloader::GetDefaultCache().GetFile(mappingJson, CACHETIME_DAY);
                         if (json == "") {
                             json = wxStandardPaths::Get().GetResourcesDir() + "/model_vendor_mapping.json";
diff --git a/xLights/xLightsMain.cpp b/xLights/xLightsMain.cpp
index d9b79b392..2231fd617 100644
--- a/xLights/xLightsMain.cpp
+++ b/xLights/xLightsMain.cpp
@@ -6723,13 +6723,13 @@ void xLightsFrame::OnMenuItem_File_NewXLightsInstance(wxCommandEvent& event)
 void xLightsFrame::OnMenuItem_Help_ReleaseNotesSelected(wxCommandEvent& event)
 {
 #ifdef __WXOSX__
-    std::string loc = "https://raw.githubusercontent.com/xLightsSequencer/xLights/" + xlights_version_string + "/README.txt";
+    std::string loc = "https://raw.githubusercontent.com/" XLIGHTS_GH_BRANCH "/" + xlights_version_string + "/README.txt";
     std::string file = CachedFileDownloader::GetDefaultCache().GetFile(wxURI(loc), CACHETIME_SESSION);
     if (file == "" || !FileExists(file)) {
         // a patch version may not have release notes so strip it off
         std::string vs = xlights_version_string;
         vs = vs.substr(0, vs.find_last_of("."));
-        loc = "https://raw.githubusercontent.com/xLightsSequencer/xLights/" + vs + "/README.txt";
+        loc = "https://raw.githubusercontent.com/" XLIGHTS_GH_BRANCH "/" + vs + "/README.txt";
     }
     ::wxLaunchDefaultBrowser(loc);
 #else
@@ -6744,7 +6744,7 @@ void xLightsFrame::OnMenuItem_Help_ReleaseNotesSelected(wxCommandEvent& event)
 
 void xLightsFrame::OnMenuItem_Help_Isue_TrackerSelected(wxCommandEvent& event)
 {
-    ::wxLaunchDefaultBrowser("https://github.com/xLightsSequencer/xLights/issues");
+    ::wxLaunchDefaultBrowser("https://github.com/" XLIGHTS_GH_BRANCH "/issues");
 }
 
 void xLightsFrame::OnMenuItem_Help_FacebookSelected(wxCommandEvent& event)
@@ -8651,7 +8651,7 @@ bool xLightsFrame::CheckForUpdate(int maxRetries, bool canSkipUpdates, bool show
 #ifdef LINUX
     wxString hostname = wxT("www.adebenham.com");
     wxString path = wxT("/wp-content/uploads/xlights/latest.php");
-    wxString downloadUrl = wxT("https://github.com/xLightsSequencer/xLights/releases/latest");
+    wxString downloadUrl = wxT("https://github.com/" XLIGHTS_GH_BRANCH "/releases/latest");
     MenuItem_Update->Enable(true);
 #else
 #ifdef __WXOSX__
diff --git a/xLights/xLightsVersion.h b/xLights/xLightsVersion.h
index 5b5f3d7ed..876be8fb5 100644
--- a/xLights/xLightsVersion.h
+++ b/xLights/xLightsVersion.h
@@ -20,11 +20,15 @@ static const wxString xlights_version_string  = "2024.11.1";
 #ifdef DEBUG
 static const wxString xlights_qualifier       = " DEBUG";
 #else
-static const wxString xlights_qualifier       = ""; // " BETA,ALPHA,PROD";
+static const wxString xlights_qualifier       = " MoC"; // " BETA,ALPHA,PROD";
 #endif
 static const wxString xlights_build_date      = __DATE__;
 	
 const wxString &GetBitness();
 wxString GetDisplayVersionString();
 
+#ifndef XLIGHTS_GH_BRANCH
+#define XLIGHTS_GH_BRANCH "merryoncherry/xLights_arch_tree_models"
+#endif
+
 #define WXWIDGETS_VERSION "33"
